src/Algebra/Ordered/Bundles.agda

```diff
@@ -0,0 +1,223 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Definitions of ordered algebraic structures like promonoids and
+-- posemigroups (packed in records together with sets, orders,
+-- operations, etc.)
+------------------------------------------------------------------------
+
+-- The contents of this module should be accessed via `Algebra.Order`.
+
+{-# OPTIONS --without-K --safe #-}
+
+module Algebra.Ordered.Bundles where
+
+open import Algebra.Core
+open import Algebra.Bundles
+open import Algebra.Ordered.Structures
+open import Level using (suc; _⊔_)
+open import Relation.Binary using (Rel)
+open import Relation.Binary.Bundles using (Preorder; Poset)
+
+------------------------------------------------------------------------
+-- Bundles of preordered structures
+
+-- Preordered semigroups (prosemigroups)
+
+record Prosemigroup c ℓ₁ ℓ₂ : Set (suc (c ⊔ ℓ₁ ⊔ ℓ₂)) where
+  infix  4 _≈_ _∼_
+  infixl 7 _∙_
+  field
+    Carrier        : Set c
+    _≈_            : Rel Carrier ℓ₁  -- The underlying equality.
+    _∼_            : Rel Carrier ℓ₂  -- The preorder.
```

Comment:
Could we use a symbol more suggestive of ordering for the preorder relation?

Reply:
Yea, I agree. I used `_∼_` to be consistent with the notation used in `Preorder` from `Relation.Binary.Bundles`, but I'd prefer `_≤_`.

Reply:
Nope!

Reply:
Fixed.